### ISSUE TITLE
Formating Node Labels

### ISSUE THREAD

**Issue by AkshatSudheshwar** [ricklupton/floweaver#71]
Hi,

I am currently trying to use floWeaver for mapping waste flows at a macroscopic level:
![image](https://user-images.githubusercontent.com/40687135/53497242-0ede2d80-3aa4-11e9-86d3-ae24205ee832.png)

I have to incorporate many more flows in this Sankey and I was hoping to somehow increase the font size of the node labels. Additionally, since some nodes have been created purely for facilitation of the sankey format, is it possible to hide/suppress the node labels without defining the nodes as waypoints (eg: nodes['level2to4_biowaste'] = Waypoint(title=''))?

**Comment by ricklupton** [comment]
Hi Akshat,

Thanks for getting in touch. floWeaver uses [ipysankeywidget](https://github.com/ricklupton/ipysankeywidget) to draw the Sankey diagrams, so you can adjust the styling as shown in the examples there ([see the bit about CSS in this section](https://nbviewer.jupyter.org/github/ricklupton/ipysankeywidget/blob/master/examples/More%20examples.ipynb#Styling)).

You could try a cell like this:
```ipython
%%html
<style>
.sankey .node {
    font-size: 20pt;
}
</style>
```

> Additionally, since some nodes have been created just for the purpose of formatting, is it possible to hide/suppress the node labels without defining the nodes as waypoints (eg: nodes['level2to4_biowaste'] = Waypoint(title=''))?

I'm not sure exactly what you mean here. It should be possible to hide the title by passing `title=''` to both `Waypoint`s and `ProcessGroup`s. Does that help?

**Comment by AkshatSudheshwar** [comment]
Hi Rick,

Thank you so much for your help with the font size issue.

To hide the node titles, currently I am using the following method to hide the node titles:

===================================================================
Level_2 = Partition.Simple('process', [**('',['consumer_2']),**
                                      'MSW_collection_CHPhigh_2', 'MSW_collection_CHPlow_2',
                                       'MSW_collection_DH_2','MSW_collection_EP_2',
                                       'MSW_collection_PS_2', **('  ',['direct_deliveries_tf_2']) ])**

nodes = { 
'Level2': ProcessGroup(['consumer_2','direct_deliveries_tf_2',
                            'MSW_collection_CHPhigh_2', 'MSW_collection_CHPlow_2',
                            'MSW_collection_DH_2','MSW_collection_EP_2',
                            'MSW_collection_PS_2'])
}

nodes['Level2'].partition = Level_2

===================================================================

For now I am defining groups in the partition.simple with titles as ''; however since the nodes with the same titles are grouped, therefore I will have to add the number of spaces within the title definition (because I wish to keep consumer_2 and direct_deliveries_tf_2 separate in the Sankey, the former is defined as '' whereas the latter as '_'). This is not very suitable because I will have to define titles with a different number of spaces for around 50-60 nodes.

If there is a particular syntax to define the title in the ProcessGroup definition, I would really appreciate your help with the same.

Thank you so much again!

P.S.  I saw that it is possible to define the title and style (shown below) but I am could not figure out how to use it within the framework of floweaver. 

![image](https://user-images.githubusercontent.com/40687135/53585836-280bda80-3b87-11e9-8b8e-f1574c773bd3.png)

**Comment by ricklupton** [comment]
I see -- no, at the moment there's no way to separate the *id* of the partition group (which must be unique, hence the multiple spaces) from the *title* displayed on the Sankey diagram. We should update the Partition code to make this possible.

A workaround might be to use multiple ProcessGroups?

```python
Level_2 = Partition.Simple('process', [('',['consumer_2']),
'MSW_collection_CHPhigh_2', 'MSW_collection_CHPlow_2',
'MSW_collection_DH_2','MSW_collection_EP_2',
'MSW_collection_PS_2', (' ',['direct_deliveries_tf_2']) ])

nodes = {
    'Level2_consumer': ProcessGroup(['consumer_2'], title=''),
    'Level2_direct_deliveries': ProcessGroup(['direct_deliveries_tf_2'], title=''),
    'Level2': ProcessGroup(['MSW_collection_CHPhigh_2', 'MSW_collection_CHPlow_2',
                            'MSW_collection_DH_2','MSW_collection_EP_2',
                            'MSW_collection_PS_2'],
                            partition=Level_2)
}
```

You would need to replace every
```python
Bundle(..., 'Level2'),
```
with
```python
Bundle(..., 'Level2_consumer'),
Bundle(..., 'Level2_direct_deliveries'),
Bundle(..., 'Level2'),
```
and vice versa for the flows leaving Level2.

**Comment by AkshatSudheshwar** [comment]
Yes, I know using multiple ProcessGroups is an option; in fact, that was present code in the originally. However, then the number of bundles will became too large to manage and connect. So we decided to go ahead with the multiple spaces option. 

Thank ou again for your help though :)
